src/main/java/Duke.java

```diff
@@ -1,10 +1,249 @@
+            int eventTimeIdx = userInput.indexOf("/");
+            String eventDescription = getEventFromUserInput(userInput,eventTimeIdx);
+            String eventTime = getEventTimeFromUserInput(userInput,eventTimeIdx);
+
+            // add Event task to List
+            addEventToList(eventDescription,eventTime);
+        }
+        else if(isDeadline(userInput)){
+            int dueDateIdx = userInput.indexOf("/");
+            String deadlineDescription = getDeadlineFromUserInput(userInput,dueDateIdx);
+            String dueDate = getDueDateFromUserInput(userInput,dueDateIdx);
+
+            // add Deadline task to List
+            addDeadlineToList(deadlineDescription,dueDate);
+        }
+    }
+
+    //method marks task in list with taskNumber as done.
+    public static void markTaskAsDone(int taskNumber){
+        // mark task with taskNumber as done.
+        taskList[taskNumber-1].markAsDone();
+        System.out.println("Great Job! I've marked the following task as done:");
+        // display updated task entry in list.
+        System.out.println(taskList[taskNumber-1]);
+        displayLine();
+    }
+
+    //method marks task in list with taskNumber as not yet done.
+    public static void unmarkTaskAsDone(int taskNumber){
+        // mark task with taskNumber as yet to be done.
+        taskList[taskNumber-1].unmarkAsDone();
+        System.out.println("Ok, I've marked the following task as yet to be done:");
+        // display updated task entry in list.
+        System.out.println(taskList[taskNumber-1]);
+        displayLine();
+    }
+
+    public static boolean isMarkCommand(String userInput){
+        if(userInput.length()>=4){
+            // return true if first 4 letters of userInput spell "mark", else false
+            return userInput.substring(0,4).equals("mark");
+        }
+        return false;
+    }
+
+    public static boolean isUnmarkCommand(String userInput){
+        if(userInput.length()>=6){
+            // return true if first 6 letters of userInput spell "mark", else false
+            return userInput.substring(0,6).equals("unmark");
+        }
+        return false;
+    }
+
+    public static boolean isListCommand(String userInput){
+        // return true if command is list, else false
+        return userInput.equals("list");
+    }
+
+    public static int getTaskNumber(String userInput){
+        return Integer.parseInt(userInput.split(" ")[1]);
+    }
+
+    private static String getUserInput() {
+        Scanner in = new Scanner(System.in);
+        String userInput = in.nextLine();
+        displayLine();
+        return userInput;
+    }
+
+    // method runs main echo functionality of duke.
+    public static void echo(){
+        while(true){
+            //read input from user.
+            String userInput = getUserInput();
+
+            // Check userInput for respective command.
+
+            // Exit if "bye" is entered by user.
+            if(userInput.equals("bye")) {
```

Comment:
Good that you have avoided deep nesting in your code with the help of your other methods.